adminSiteClient/GdocsPreviewPage.tsx

```diff
@@ -0,0 +1,229 @@
+        const fetchOriginalGdoc = async () => {
+            const originalGdocJson = (await admin.getJSON(
+                `/api/gdocs/${id}`
+            )) as OwidArticleTypeJSON
+
+            setOriginalGdoc(getArticleFromJSON(originalGdocJson))
+        }
+        fetchOriginalGdoc()
+    }, [admin, id])
+
+    const hasWarnings =
+        errors?.some((error) => error.type === ErrorMessageType.Warning) ??
+        false
+    const hasErrors =
+        errors?.some((error) => error.type === ErrorMessageType.Error) ?? false
+
+    const confirmPublish = async () => {
+        if (!gdoc) return
+
+        Modal.confirm({
+            title: "Are you sure you want to publish this article?",
+            content:
+                hasWarnings &&
+                errors?.map((error, i) => <div key={i}>{error.message}</div>),
+            okType: hasWarnings ? "danger" : "primary",
+            okText: "Publish now",
+            cancelText: "Cancel",
+            onOk: doPublish,
+        })
+    }
+
+    const doPublish = async () => {
+        if (!gdoc) return
+        const publishedGdoc = await store.publish(gdoc)
+
+        setGdoc(publishedGdoc)
+        setOriginalGdoc(publishedGdoc)
+        openSuccessNotification()
+    }
+
+    const doUnpublish = async () => {
+        if (!gdoc) return
+        const unpublishedGdoc = await store.unpublish(gdoc)
+
+        setGdoc(unpublishedGdoc)
+        setOriginalGdoc(unpublishedGdoc)
+        openSuccessNotification()
+    }
+
+    const onDelete = async () => {
+        if (!gdoc) return
+        await store.delete(gdoc)
+        history.push("/gdocs")
+    }
+
+    const onSettingsClose = () => {
+        setSettingsOpen(false)
+    }
+
+    // Handle errors and validation status
+    useEffect(() => {
+        if (!gdoc) return
+        const errors = getErrors(gdoc)
+        setErrors(errors)
+    }, [gdoc])
+
+    return gdoc ? (
+        <AdminLayout
+            title={`Previewing ${gdoc.content.title}`}
+            noSidebar
+            fixedNav={false}
+        >
+            <main className="GdocsEditPage">
+                <Row
+                    justify="space-between"
+                    align="middle"
+                    gutter={[16, 8]}
+                    className={`p-3 admin-bar ${
+                        gdoc.published ? "published" : "draft"
+                    }`}
+                >
+                    <Col flex={1}>
+                        <Space>
+                            <Button onClick={() => history.push("/gdocs")}>
+                                <FontAwesomeIcon icon={faAngleLeft} />
+                            </Button>
+                            <Typography.Title className="mb-0" level={4}>
+                                {gdoc.content.title}
+                            </Typography.Title>
+                            [ <GdocsEditLink gdoc={gdoc} /> ]
+                            <div>
+                                {!gdoc.published && (
```

Comment:
I _think_ React takes care of not rendering `false`.
It's only something like `list.length && list.map(...)` that we need to watch out for, because it can render `0`.

Reply:
are you talking about the case where the first operand could evaluate to 0 and "0" would be printed out? In this case gdoc.published is a boolean, so this works as expected (as opposed to someArray.length && Component, which could indeed print out "0" if someArray was empty). Let me know if I misunderstood.

Reply:
Ah yes, that's exactly what I was saying actually.
It's totally fine here, and I was just explaining the general case where we have to watch out 🙌🏻

Reply:
ah sorry for the stereo, I was commenting through VSCode and your comment hadn't shown there yet 😅

Reply:
🙊